contracts/TracerPerpetualSwaps.sol

```diff
@@ -198,55 +197,38 @@ contract TracerPerpetualSwaps is
      */
     function executeTrade(
         Perpetuals.Order memory order1,
-        Perpetuals.Order memory order2,
-        uint256 fillAmount
+        Perpetuals.Order memory order2
     ) internal {
-        // fill amount > 0. Overflow occurs when fillAmount > 2^256 - 1
-        int256 _fillAmount = fillAmount.toInt256();
-        int256 quoteChange =
-            PRBMathUD60x18.mul(fillAmount, order1.price).toInt256();
-
-        // Update account states
+        // Retrieve account state
         Balances.Account storage account1 = balances[order1.maker];
         Balances.Account storage account2 = balances[order2.maker];
 
+        // Construct `Trade` types suitable for use with LibBalances
+        (Balances.Trade memory trade1, Balances.Trade memory trade2) =
+            (
+                Balances.Trade(order1.price, order1.amount, order1.side),
+                Balances.Trade(order2.price, order2.amount, order2.side)
+            );
+
+        // Calculate new account state
+        (Balances.Position memory newPos1, Balances.Position memory newPos2) =
+            (
+                Balances.applyTrade(account1.position, trade1, feeRate),
```

Comment:
`applyTrade` applies the entire trade amount to the position, not the match amount. Some how we need to either factor `fillAmount` into this or alter the actual trade object to have the correct amount

Reply:
Consult c1c2220